include/alpaka/pltf/PltfGenericSycl.hpp

```diff
@@ -18,89 +18,164 @@
 
 #    include <CL/sycl.hpp>
 
-#    include <iostream>
+#    include <cstddef>
+#    include <exception>
+#    include <functional>
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+#        include <iostream>
+#    endif
+#    include <mutex>
+#    include <optional>
 #    include <sstream>
 #    include <stdexcept>
 #    include <vector>
 
 namespace alpaka::experimental
 {
     //! The SYCL device manager.
-    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl>
+    template<typename TSelector>
+    class PltfGenericSycl : public concepts::Implements<ConceptPltf, PltfGenericSycl<TSelector>>
     {
+        using RecursiveLock = std::unique_lock<std::recursive_mutex>;
```

Comment:
Why not?
```suggestion
        using RecursiveLock = std::scoped_lock<std::recursive_mutex>;
```

Reply:
What would be the benefit of using `scoped_lock` in this case?

Reply:
It's a less capable, simpler mechanism. It is the C++17 replacement of ` lock_guard`. `unique_lock` allows you to unlock manually as well, so has more complexity. `scoped_lock`/`lock_guard` are pure RAII solutions.